### DISCUSSION TITLE
Does window title work in v2

### DISCUSSION THREAD

**Discussion by sushibagel** [Spawnova/ShinsImageScanClass#30] [Q&A]
Let me start by thanking you for sharing this. It seems like a cool class and I can really see it coming in handy for my projects. 

That being said in experimenting with using the class I can't seem to get it to work correctly with a targeted window. I used the code below and it didn't work. However, if I take "ahk_exe Code.exe" out it works perfectly so long as the window is active. I did also try just the window title and still no luck. 

Thanks in advance for any help. 

`
VSCodeScan := ShinsImageScanClass("ahk_exe Code.exe")
VSCodeScan.AutoUpdate := 0  ;only want to update the buffer once then perform many scans
VSCodeScan.UseControlClick := 1

#i::
{
    VSCodeScan.Update() ;update buffer
    If VSCodeScan.Image("testimage.png")
    {
        TestGui := Gui(,"My Test Gui")
        TestGui.Add("Text",,"This is text in my Gui!")
        TestGui.Opt("-Caption AlwaysOnTop")
        TestGui.Show
    }
}

#IncludeAgain ImageScanClass.ahk
#IncludeAgain OverlayClass.ahk`

**Comment by Spawnova** [comment]
Hey there, it does work in the way that you describe, however my class uses BitBlt to copy the pixel data and it is not supported by all applications, particularly it is often not always compatible with applications using the gpu in a specific way

To test if that is the case you can run the above code but dump the pixel data to a png image to check to see if it looks corect or not
```ahk
#IncludeAgain ImageScanClass.ahk
VSCodeScan := ShinsImageScanClass("ahk_exe Code.exe")
VSCodeScan.SaveImage("vscode.png")
exitapp
```
if that image is all black, or it doesn't look right, then it's probably not compatible in target window mode, by not specifying a window title the class defaults to desktop mode, which is slower and can't work on background windows =[

some applications can be configured to disable hardware acceleration, and that can fix the problem though if possible

alternatively for windows with controls there are other ahk libraries that may prove more useful for automating them, I don't know off the top of my head the names, but I'm sure a google search should come up with something

**Reply by sushibagel** [reply]
Thanks for the quick reply. It seems you were correct and that it just wasn't a compatible application. Thanks for your help.
